pkg/env/env_test.go

```diff
@@ -0,0 +1,168 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package env
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestMergeEnvVars(t *testing.T) {
+	type args struct {
+		new             []corev1.EnvVar
+		into            []corev1.EnvVar
+		overwriteValues bool
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []corev1.EnvVar
+		wantErr bool
+	}{
+		{
+			name: "should not fail with nil inputs",
+			args: args{
+				new:             nil,
+				into:            nil,
+				overwriteValues: false,
+			},
+			want:    []corev1.EnvVar{},
+			wantErr: false,
+		},
+		{
+			name: "duplicate names should fail with overwriteValues false",
```

Comment:
This is interesting, because in your test data, an overwrite would not actually happen because the value in `new` equals the value in `into`. I have not made my mind on how want the behavior.

Reply:
I can change the Value of the env var to make this particular test more clear, but I was only testing for getting an error when a name collision is encountered and overwriteValues is false.